### PR TITLE
Remove the deprecated Test.srcdir and Test.datadir attributes

### PR DESCRIPTION
**Pull request by clebergnu**
These have been deprecated for a while, and are now past due their removal, so let's do it now.

### CODE REVIEW

File: docs/source/WritingTests.rst [hunk 1]
```diff
@@ -227,10 +227,12 @@ you intend to create it.
          for that specific test and execution conditions (such as with or
          without variants).  Look for "Test data directories" in the test logs.
 
-.. note:: An older API, :attr:`avocado.core.test.Test.datadir`, allows access
+.. note:: An extint API, ``avocado.core.test.Test.datadir``, allowed access
```
Comment (ldoktor) [inline docs/source/WritingTests.rst]: I guess you meant `extinct`, right?

Comment (ldoktor) [reply docs/source/WritingTests.rst]: Perhaps I'd use `superseded` or `deprecated` to make it easier to understand...

File: selftests/unit/test_test.py [hunk 2]
```diff
@@ -98,20 +98,20 @@ def check(uid, name, variant, exp_logdir):
 
     def test_data_dir(self):
         """
-        Tests that a valid datadir exists following the test filename
+        Tests that a valid data dir exists following the test filename
```
Comment (ldoktor) [inline selftests/unit/test_test.py]: Well the docstring is not really accurate. How about:

```
Checks that `get_data` reports `data dir` for test name of the maximum allowed length.
```

File: selftests/unit/test_test.py [hunk 3]
```diff
@@ -98,20 +98,20 @@ def check(uid, name, variant, exp_logdir):
...
-                         tst.datadir)
+                         tst.get_data('', 'file', False))
 
     def test_no_data_dir(self):
         """
         Tests that with a filename too long, no datadir is possible
         """
         above_limit_name = os.path.join(self.tmpdir, "a" * 251)
         tst = self._get_fake_filename_test(above_limit_name)
-        self.assertFalse(tst.datadir)
```
Comment (ldoktor) [inline selftests/unit/test_test.py]: And here the docstring might by synced to say `Checks "get_data" won't report fs-unfriendly data dir name.`

File: docs/source/WritingTests.rst (hunk 1 above)
Comment (clebergnu) [reply docs/source/WritingTests.rst]: Not deprecated... as deprecated it was *before* this.  Let's use "A previously existing API, now removed, ... "

**Comment by clebergnu** [comment]
Closing in favor of #2661

**Pull request closed without merging**